### DISCUSSION TITLE
how to work with setHover?

### DISCUSSION THREAD

**Discussion by wujunbao1963** [marcofugaro/three-projected-material#49] [Q&A] [answered]
In [this](https://codesandbox.io/s/basic-projected-material-test-forked-n22sng?file=/src/index.js:1560-1568)  script, when use setSelected(e) or setHover(true), the projected texture just disappeared, how to fix it to keep the texture always being there?
Thanks for Projected Material, it helps me a lot, making things easier.

**Comment by marcofugaro** [comment, marked as answer]
Hey, not sure why about `extend()`, but you can use ProjectedMaterial in r3f like this

```js
const ProjectedMaterial = React.forwardRef(
  ({ camera, texture, ...props }, ref) => {
    const material = useMemo(
      () => new ProjectedMaterialImpl({ camera, texture }),
      [camera, texture]
    );
    return (
      <primitive object={material} ref={ref} attach="material" {...props} />
    );
  }
);
```

See the full demo here
https://codesandbox.io/s/basic-projected-material-test-forked-07u45e?file=/src/index.js:616-928

**Comment by wujunbao1963** [comment]
hi marcofugaro, it works! thank you so much.
